.github/workflows/build.yml

```
@@ -82,11 +82,15 @@ jobs:
           # CPython, so CIFuzz should be run only for code that is likely to be
           # merged into the main branch; compatibility with older branches may
           # be broken.
-          if [ "$GITHUB_BASE_REF" = "main" ]; then
+          FUZZ_RELEVANT_FILES='(\.c$|\.h$|\.cpp$|^configure$|^\.github/workflows/build\.yml$|^Modules/_xxtestfuzz)'
+          if [ "$GITHUB_BASE_REF" = "main" ] && [ "$(git diff --name-only origin/$GITHUB_BASE_REF.. | grep -qvE $FUZZ_RELEVANT_FILES; echo $?)" -eq 1 ]; then
```

Do I get this logic right?

First part:

* `[ "$GITHUB_BASE_REF" = "main" ]` - this is a PR

Second part:

* `git diff --name-only origin/$GITHUB_BASE_REF..` - lists files changed compared with `main`

* `grep -qvE $FUZZ_RELEVANT_FILES` - checks the changed files are NOT of the relevant type

* `-eq 1` - error code 1, so true only when the files ARE found

---

Instead of checking no match is false: `(...; echo $?)" -eq 1`

Can we check for a match, something along the lines of this?

```suggestion
          if [ "$GITHUB_BASE_REF" = "main" ] && [ "$(git diff --name-only origin/$GITHUB_BASE_REF.. | grep -qE $FUZZ_RELEVANT_FILES)" ]; then
```

---

When using `grep -q` in `$()`, we have to reply on exit code, because no output is produced.

So:

```bash
(.venv) ~/Desktop/cpython  fix-ci-fuzz-build ✔                                            
» echo 'abc' | grep -qE 'b'; echo $?
0
                                                                                           
(.venv) ~/Desktop/cpython  fix-ci-fuzz-build ✔                                            
» echo 'abc' | grep -qE 'y'; echo $?
1
```